### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by hakichris**
In this branch, we apply validation to the database

### CODE REVIEW

File: spec/models/like_spec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+RSpec.describe Like, type: :model do
+  subject { Like.new(author_id: 3, post_id: 3) }
+
+  context 'it should exist' do
+    it 'should exist as instance' do
+      expect(subject).to be_instance_of Like
+    end
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline spec/models/like_spec.rb:11]: - [ ] kindly add test for the method in your like mode class `updatelikescounter`

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      subject.text = nil
+      expect(subject).to_not be_valid
+    end
+  end
+
+  context 'testing scopes' do
+    it 'should return a number' do
+      expect(subject.recentfivecomments).to_not eq 0
+    end
+  end
```
Comment (VICTOR-CHUKWU) [inline spec/models/post_spec.rb:23]: - [ ] kindly add a test for all the methods in your post model e.g `updatepostscounter`

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    it 'should return a number' do
+      expect(subject.postscounter).to_not eq 0
+    end
+  end
+
+  context 'testing scopes' do
+    it 'should return 3 examples' do
+      expect(subject.recentthreeposts).to_not eq 0
+    end
+  end
```
Comment (VICTOR-CHUKWU) [inline spec/models/user_spec.rb:30]: - [ ] kindly add a test for `recentthreeposts` method in your user model

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  subject { Comment.new(author_id: 3, post_id: 3) }
+
+  context 'it should exist' do
+    it 'should exist as instance' do
+      expect(subject).to be_instance_of Comment
+    end
+  end
```
Comment (VICTOR-CHUKWU) [inline spec/models/comment_spec.rb:10]: - [ ] kindly add test for `updatecommentscounter` method in your comment model

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @hakichris  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb (hunk 3 above)
Comment (hakichris) [reply spec/models/user_spec.rb]: @VICTOR-CHUKWU  done

File: spec/models/post_spec.rb (hunk 2 above)
Comment (hakichris) [reply spec/models/post_spec.rb]: @VICTOR-CHUKWU  done

File: spec/models/like_spec.rb (hunk 1 above)
Comment (hakichris) [reply spec/models/like_spec.rb]: @VICTOR-CHUKWU  done

File: spec/models/comment_spec.rb (hunk 4 above)
Comment (hakichris) [reply spec/models/comment_spec.rb]: @VICTOR-CHUKWU  done

**Review by DammyShittu — APPROVED** [review]
Hi @hakichris,

## APPROVED 🟢

## Highlights
✔️ Passing tests

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**